pkg/repository/v1/sqlcv1/concurrency.sql

```diff
@@ -305,18 +306,13 @@ WITH filled_parent_slots AS (
         cs.*
     FROM
         v1_concurrency_slot cs
+    JOIN
+        eligible_slots_per_group es ON cs.task_id = es.task_id
     WHERE
-        (task_inserted_at, task_id, task_retry_count, tenant_id, strategy_id) IN (
```

Comment:
Noticed very high query latency with this high cardinality `IN` statement and it seems like this is a pattern we should avoid when the right side of the join has a lot of rows. Anecdotally this improved performance a lot.